### PR TITLE
[Issue #26] Make pre_render compatible with all children of ActiveRecord::Relation

### PR DESCRIPTION
**Pull request by ryanmccarthypdx**
Checklist:

* [x] I have updated the necessary documentation
* [x] I have signed off all my commits as required by [DCO](https://github.com/procore-oss/blueprinter-activerecord/blob/main/CONTRIBUTING.md)
* [x] My build is green

### CODE REVIEW

**Comment by jhollinger** [comment]
Have to study this more to make sure, but at a glance it seems like the perfect idea. Nice!

File: test/nested_render_test.rb [hunk 1]
```diff
@@ -40,7 +43,15 @@ def test_queries_with_auto
     assert_equal [
       'SELECT "projects".* FROM "projects"',
       'SELECT "customers".* FROM "customers" WHERE "customers"."id" IN (?, ?)',
-      'SELECT "widgets".* FROM "widgets" WHERE "widgets"."project_id" IN (?, ?)',
+      'SELECT "widgets".* FROM "widgets" WHERE "widgets"."project_id" IN (?, ?, ?)',
+    ], @queries
+  end
+
+  def test_queries_for_collection_proxies
+    ProjectBlueprint.render(@test_customer.projects.strict_loading, view: :extended_plus_with_widgets)
```
Comment (jhollinger) [inline test/nested_render_test.rb:51]: I don't think this line isn't testing exactly what we want. When you add the `strict_loading` call, it changes it from `ActiveRecord::Associations::CollectionProxy` to `ActiveRecord::AssociationRelation`, which was one of the "approved" classes prior to this change.

**Review by jhollinger — CHANGES_REQUESTED** [review]
Looks good overall, just one quibble about the tests.

File: test/nested_render_test.rb (hunk 1 above)
Comment (ryanmccarthypdx) [reply test/nested_render_test.rb]: Good catch

**Review by jhollinger — APPROVED** [review]

**Pull request merged**